cmd/ci-op-configs-mirror/main.go

```diff
@@ -0,0 +1,130 @@
+)
+
+type options struct {
+	dryRun  bool
+	private bool
+
+	configDir string
+	toOrg     string
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.configDir, "config-path", "", "Path to directory containing ci-operator configurations")
+	fs.StringVar(&o.toOrg, "to-org", "", "Name of the organization which the ci-operator configuration files will be mirrored")
+
+	fs.BoolVar(&o.dryRun, "dry-run", false, "Whether to actually generate the new ci-operator configuration files")
+	fs.BoolVar(&o.private, "private", false, "Generate a prowgen configuration file for a private organization")
+
+	fs.Parse(os.Args[1:])
+	return o
+}
+
+func (o *options) validate() error {
+	if len(o.configDir) == 0 {
+		return errors.New("--config-path is not defined")
+	}
+
+	if len(o.toOrg) == 0 {
+		return errors.New("--to-org is not defined")
+	}
+	return nil
+}
+
+func main() {
+	o := gatherOptions()
+	if err := o.validate(); err != nil {
+		logrus.WithError(err).Fatal("Invalid option")
+	}
+
+	go func() {
+		interrupts.WaitForGracefulShutdown()
+		os.Exit(1)
+	}()
+
+	dryBuff := &bytes.Buffer{}
+	callback := func(rbc *api.ReleaseBuildConfiguration, repoInfo *config.Info) error {
+		logger := logrus.WithFields(logrus.Fields{"org": repoInfo.Org, "repo": repoInfo.Repo, "branch": repoInfo.Branch})
+
+		if !promotion.BuildsOfficialImages(rbc) {
+			logger.Warn("Skiping...")
+			return nil
+		}
+
+		repoInfo.Org = o.toOrg
+
+		logger.Info("Processing...")
+
+		// Disable promotion
+		rbc.PromotionConfiguration.Disabled = true
+		rbc.PromotionConfiguration.Name = privatePromotionIS
+
+		data, err := yaml.Marshal(rbc)
+		if err != nil {
+			logger.WithError(err).Fatal("couldn't marshal ci-operator's configuration")
+		}
+
+		if o.dryRun {
+			dryBuff.WriteString(fmt.Sprintf("%s:\n%v\n", repoInfo.RelativePath(), string(data)))
+			return nil
+		}
+
+		dataWithInfo := config.DataWithInfo{
+			Configuration: *rbc,
+			Info:          *repoInfo,
+		}
+
+		if err := dataWithInfo.CommitTo(o.configDir); err != nil {
+			logger.WithError(err).Fatal("couldn't create the configuration file")
+		}
+
+		return nil
+	}
+
+	if err := config.OperateOnCIOperatorConfigDir(o.configDir, callback); err != nil {
+		logrus.WithError(err).Fatal("error while generating the ci-operator configuration files")
+	}
+
+	if o.private {
+		prowgenConfig := config.Prowgen{Private: true}
+		data, err := yaml.Marshal(prowgenConfig)
+		if err != nil {
+			logrus.WithError(err).Fatal("couldn't marshal prowgen configuration")
+		}
+
+		prowgenConfigPath := filepath.Join(o.toOrg, config.ProwgenFile)
+		if o.dryRun {
+			dryBuff.WriteString(fmt.Sprintf("%s:\n%v", prowgenConfigPath, string(data)))
+			fmt.Printf("%s", dryBuff)
+		} else {
+			if err := ioutil.WriteFile(filepath.Join(o.configDir, prowgenConfigPath), data, 0755); err != nil {
+				logrus.WithError(err).Fatal("error while writing to file")
+			}
+		}
+	}
```

Comment:
I think there is no reason why the tool should be doing this. We already have that file (https://github.com/openshift/release/tree/master/ci-operator/config/openshift-priv) IMO we don't need an 1) additional option 2) additional code. It just increases complexity and adds some coupling.

Reply:
Ok, I'll remove it. Less code is always better!